plugins/techdocs-addons-test-utils/src/test-utils.tsx

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactElement, Fragment } from 'react';
+
+// Shadow DOM support for the simple and complete DOM testing utilities
+// https://github.com/testing-library/dom-testing-library/issues/742#issuecomment-674987855
+import { screen } from 'testing-library__dom';
+import { renderToStaticMarkup } from 'react-dom/server';
+import { Route } from 'react-router-dom';
+import { act, render } from '@testing-library/react';
+
+import {
+  wrapInTestApp,
+  TestApiProvider,
+  TestApiProviderProps,
+} from '@backstage/test-utils';
+import { FlatRoutes } from '@backstage/core-app-api';
+
+import {
+  TechDocsAddons,
+  techdocsApiRef,
+  TechDocsEntityMetadata,
+  TechDocsMetadata,
+  techdocsStorageApiRef,
+} from '@backstage/plugin-techdocs-react';
+import { TechDocsReaderPage, techdocsPlugin } from '@backstage/plugin-techdocs';
+import { catalogPlugin } from '@backstage/plugin-catalog';
+import { searchApiRef } from '@backstage/plugin-search-react';
+
+/**
+ * @public
+ */
+export type RecursivePartial<T> = {
+  [P in keyof T]?: RecursivePartial<T[P]>;
+};
+
+const techdocsApi = {
+  getTechDocsMetadata: jest.fn(),
+  getEntityMetadata: jest.fn(),
+};
+
+const techdocsStorageApi = {
+  getApiOrigin: jest.fn(),
+  getEntityDocs: jest.fn(),
+  syncEntityDocs: jest.fn(),
+};
+
+const searchApi = {
+  query: jest.fn().mockResolvedValue({ results: [] }),
+};
+
+/**
+ * @public
+ */
+export type Apis = TestApiProviderProps<any>['apis'];
```

Comment:
Very unfortunate to lose the type checking by using `any` here. Also a very generic type so would rather avoid it tbh